mne/tests/test_report.py

```diff
@@ -337,4 +335,81 @@ def test_open_report():
     open_report(hdf5, subjects_dir=subjects_dir)  # This should work
 
 
+def test_remove():
+    """Test removing figures from a report."""
+    r = Report()
+    r.add_figs_to_section(fig1, 'figure1', 'mysection')
+    r.add_figs_to_section(fig1, 'figure1', 'othersection')
+    r.add_figs_to_section(fig2, 'figure1', 'mysection')
+    r.add_figs_to_section(fig2, 'figure2', 'mysection')
+
+    # Test removal by fname
+    r2 = copy.deepcopy(r)
+    removed_fname = r2.remove(fname='figure1-#-report_mysection-#-custom')
```

Comment:
Hmm ... doesn't the report do a reordering inside? At least if you used `parse_folder` ...

Reply:
Also, why would caption + section not be enough for narrowing down?

Reply:
Custom figures have no re-ordering done to them, no. The `self.html` list is just dumped to disk.

> Also, why would caption + section not be enough for narrowing down?

Because there can be multiple figures with the same caption. The caption is not a unique identifier for a figure.

Reply:
also see here: https://github.com/mne-tools/mne-python/blob/master/mne/report.py#L1547.

Basically, the order in which you add figures is not necessarily the order in which they appear on the report.

Reply:
ok, then I agree that search by caption is the best we can do.